src/SUnitRange.jl

```diff
@@ -0,0 +1,12 @@
+# `Start` and `End` should be `Int`
+struct SUnitRange{Start,End} <: StaticVector{Int}
+end
+
+@pure Size(::SUnitRange{Start, End}) where {Start,End} = Size(End-Start+1)
+
+@pure @propagate_inbounds function getindex(x::SUnitRange{Start,End}, i::Int) where {Start, End}
+    @boundscheck if i < Start || i > End
```

Comment:
Haha thanks @timholy - these definitely are WIP. I think I'll split this off and put it on a branch for now.

These are just meant to be singleton array types, so that `indices(SVector(1,2,3)) === SUnitRange{1,3}()`. AFAICT `IdentityRanges` are meant to be a bit like a safe `:`? (How is it distinct from `start:stop`?)

I suppose that `OneTo` has been able to double-up for both purposes, for one-based arrays?